plat/common/aarch64/platform_helpers.S

```diff
@@ -57,31 +58,54 @@ func plat_report_exception
 endfunc plat_report_exception
 
 	/* -----------------------------------------------------
-	 * Placeholder function which should be redefined by
-	 * each platform.
+	 * int plat_crash_console_init(void)
+	 * Use normal console by default. Switch it to crash
+	 * mode so serial consoles become active again.
+	 * NOTE: This default implementation will only work for
+	 * crashes that occur after a normal console (marked
+	 * valid for the crash state) has been registered with
+	 * the console framework. To debug crashes that occur
+	 * earlier, the platform has to override these functions
+	 * with an implementation that initializes a console
+	 * driver with hardcoded parameters. See
+	 * docs/porting-guide.rst for more information.
 	 * -----------------------------------------------------
 	 */
 func plat_crash_console_init
+#if defined(IMAGE_BL1)
```

Comment:
The checks failed for several build configurations with the following: 
/work/workspace/trusted-firmware-build/BUILD_CONFIG/hikey960-default/MODE/Debug/label/platforms/plat/common/aarch64/platform_helpers.S:87: undefined reference to `console_switch_state'
/work/workspace/trusted-firmware-build/BUILD_CONFIG/hikey960-default/MODE/Debug/label/platforms/plat/common/aarch64/platform_helpers.S:87:(.text.plat_crash_console_init+0x8): relocation truncated to fit: R_AARCH64_CALL26 against undefined symbol `console_switch_state'

I think the issue is that this line should be:
#if defined(IMAGE_BL1) || defined(IMAGE_BL2)

This fixes the failing hikey builds - build options:
- DEBUG=1 PLAT=hikey960
- DEBUG=1 PLAT=hikey960 SPD=opteed
The impacted build configurations are:
- hikey960: make

Reply:
No, I think the condition should be:

    #if defined (MULTI_CONSOLE) && defined (IMAGE_BL1)

The other changes to the plat_crash_console_xxx functions should also be configured under MULTI_CONSOLE. Platforms shoudn't be forced to change these functions.

Reply:
Yes, sorry, this whole thing needs to be wrapped into an #if MULTI_CONSOLE_API block. Not sure at what point I forgot to do that. Fixed.